tests/contrib/sanic/test_sanic_server.py

```diff
@@ -1,100 +1,76 @@
-httpx_client = getattr(httpx, "AsyncClient", getattr(httpx, "Client"))
+RUN_SERVER_PY = os.path.abspath(os.path.join(os.path.dirname(__file__), "run_server.py"))
 
+sanic_version = tuple(map(int, sanic_version.split(".")))
 
-@pytest.fixture
-def app(tracer):
-    app = Sanic(__name__)
 
-    @tracer.wrap()
-    async def random_sleep():
-        await asyncio.sleep(random.random())
+def is_server_listening():
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(("0.0.0.0", 8000)) == 0
 
-    @app.route("/hello")
-    async def hello(request):
-        await random_sleep()
-        return json({"hello": "world"})
 
-    @app.route("/internal_error")
-    async def internal_error(request):
-        1 / 0
-
-    yield app
-
-
-@pytest.fixture
-async def sanic_http_server(app, unused_port, loop):
+@pytest.fixture(autouse=True)
+def sanic_http_server():
     """Fixture for using sanic async HTTP server rather than a asgi async server used by test client"""
-    DEFAULT_CONFIG["REGISTER"] = False
-    server = await app.create_server(debug=True, host="0.0.0.0", port=unused_port, return_asyncio_server=True)
-    yield server
-    server.close()
-    await server.wait_closed()
-
-
-@pytest.mark.asyncio
-async def test_multiple_requests_sanic_http(tracer, sanic_http_server, unused_port):
-    url = "http://0.0.0.0:{}/hello".format(unused_port)
-    async with httpx_client() as client:
-        responses = await asyncio.gather(
-            client.get(url),
-            client.get(url),
-        )
-
-    assert len(responses) == 2
-    assert [r.status_code for r in responses] == [200] * 2
-    assert [r.json() for r in responses] == [{"hello": "world"}] * 2
-
-    spans = tracer.pop_traces()
-    assert len(spans) == 2
-    assert len(spans[0]) == 2
-    assert len(spans[1]) == 2
-
-    assert spans[0][0].name == "sanic.request"
-    assert spans[0][1].name == "tests.contrib.sanic.test_sanic_server.random_sleep"
-    assert spans[0][0].parent_id is None
-    assert spans[0][1].parent_id == spans[0][0].span_id
-    assert spans[0][0].get_tag("http.status_code") == "200"
-
-    assert spans[1][0].name == "sanic.request"
-    assert spans[1][1].name == "tests.contrib.sanic.test_sanic_server.random_sleep"
-    assert spans[1][0].parent_id is None
-    assert spans[1][1].parent_id == spans[1][0].span_id
-    assert spans[1][0].get_tag("http.status_code") == "200"
-
-
-@pytest.mark.asyncio
-async def test_sanic_errors(tracer, sanic_http_server, unused_port):
-    url = "http://0.0.0.0:{}/not_found".format(unused_port)
-    async with httpx_client() as client:
-        response = await client.get(url)
-
+    env = os.environ.copy()
+    env["SANIC_PORT"] = "8000"
+    args = ["ddtrace-run", "python", RUN_SERVER_PY]
+    subp = subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, close_fds=True, env=env)
+    try:
+        # Wait up to 2 seconds for the server to be listening
```

Comment:
We have an existing mechanism/pattern for this that we could reuse: https://github.com/DataDog/dd-trace-py/blob/1dc5f8a43c5dd663ca9505ff2432a5482417e5e2/tests/contrib/django/test_django_snapshots.py#L16-L23